homeassistant/util/json.py

```diff
@@ -60,29 +61,14 @@ def save_json(
         _LOGGER.error(msg)
         raise SerializationError(msg) from error
 
-    tmp_filename = ""
-    tmp_path = os.path.split(filename)[0]
     try:
-        # Modern versions of Python tempfile create this file with mode 0o600
-        with tempfile.NamedTemporaryFile(
-            mode="w", encoding="utf-8", dir=tmp_path, delete=False
-        ) as fdesc:
-            fdesc.write(json_data)
-            tmp_filename = fdesc.name
-        if not private:
-            os.chmod(tmp_filename, 0o644)
-        os.replace(tmp_filename, filename)
+        with atomicwrites.atomic_write(filename, overwrite=True) as fp:
+            fp.write(json_data)
```

Comment:
Ideally we should also make the mod change part of the atomic operation, I think it should be possible as the new file has been created at this point and we can change its mod using `fchmod`, e.g. `os.fchmod(f.fileno(), 0o600 if private else 0o644)`

Reply:
We can use `os.chmod` from Python 3.3.
https://docs.python.org/3/library/os.html#os.chmod